File: Trabajo_8/JAVA/Actividades/Actividad_1.java
/*
package Trabajo_7.Actividades;

import java.util.Scanner;

interface Forma {
    double calcularPerimetro();
}

class Circulo implements Forma {
    private double radio;

    public Circulo(double radio) {
        this.radio = radio;
    }

    @Override
    public double calcularPerimetro() {
        return 2 * Math.PI * radio;
    }
}

class Rectangulo implements Forma {
    private double largo;
    private double ancho;

    public Rectangulo(double largo, double ancho) {
        this.largo = largo;
        this.ancho = ancho;
    }

    @Override
    public double calcularPerimetro() {
        return 2 * (largo + ancho);
    }
}

class Triangulo implements Forma {
    private double lado1;
    private double lado2;
    private double lado3;

    public Triangulo(double lado1, double lado2, double lado3) {
        this.lado1 = lado1;
        this.lado2 = lado2;
        this.lado3 = lado3;
    }

    @Override
    public double calcularPerimetro() {
        return lado1 + lado2 + lado3;
    }
}

public class Actividad_2 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Forma forma = null;

        System.out.println("Seleccione una forma (1: Círculo, 2: Rectángulo, 3: Triángulo): ");
        int opcion = scanner.nextInt();

        switch (opcion) {
            case 1:
                System.out.println("Ingrese el radio del círculo: ");
                double radio = scanner.nextDouble();
                forma = new Circulo(radio);
                break;
            case 2:
                System.out.println("Ingrese el largo y ancho del rectángulo: ");
                double largo = scanner.nextDouble();
                double ancho = scanner.nextDouble();
                forma = new Rectangulo(largo, ancho);
                break;
            case 3:
                System.out.println("Ingrese los tres lados del triángulo: ");
                double lado1 = scanner.nextDouble();
                double lado2 = scanner.nextDouble();
                double lado3 = scanner.nextDouble();
                forma = new Triangulo(lado1, lado2, lado3);
                break;
            default:
                System.out.println("Opción no válida.");
                break;
        }

        if (forma != null) {
            System.out.println("El perímetro es: " + forma.calcularPerimetro());
        }

        scanner.close();
    }
}
*/

File: Trabajo_8/CPP/Actividades/Actividad_1.cpp
#include <iostream>
using namespace std;

class Figura {
    public:
        virtual double calcularArea() const = 0;
};

class Circulo : public Figura {
    public:
        explicit Circulo(double r=1) : radio(r) {}
        double calcularArea() const override {
            return 3.14 * radio * radio;
        }

    private:
        double radio;
};

class Rectangulo : public Figura {
    public:
        explicit Rectangulo(double w=1, double h=1) : ancho(w), alto(h) {}
        double calcularArea() const override {
            return ancho * alto;
        }

    private:
        double ancho;
        double alto;
};

class Cuadrado : public Figura {
    public:
        explicit Cuadrado(double l=1) : lado(l) {}
        double calcularArea() const override {
            return lado * lado;
        }

    private:
        double lado;
};

class Pentagono : public Figura {
    public:
        explicit Pentagono(double l=1) : lado(l) {}
        double calcularArea() const override {
            return 1.72 * lado * lado;
        }

    private:
        double lado;
};

class Triangulo : public Figura {
    public:
        explicit Triangulo(double b=1, double h=1) : base(b), altura(h) {}
        double calcularArea() const override {
            return 0.5 * base * altura;
        }

    private:
        double base;
        double altura;
};


int main() {
    int opcion;
    double a, b;

    cout << "Elige una figura:\n1. Circulo\n2. Rectangulo\n3. Triangulo\n4. Cuadrado\n5. Pentagono\n-> ";
    cin >> opcion;

    switch (opcion) {
        case 1:
            cout << "Introduce el radio: ";
            cin >> a;
            cout << "El area del circulo es de " << Circulo(a).calcularArea() << endl;
            break;
        case 2:
            cout << "Introduce el ancho y la altura: ";
            cin >> a >> b;
            cout << "El area del rectangulo es de " << Rectangulo(a, b).calcularArea() << endl;
            break;
        case 3:
            cout << "Introduce la base y la altura: ";
            cin >> a >> b;
            cout << "El area del triangulo es de " << Triangulo(a, b).calcularArea() << endl;
            break;
        case 4:
            cout << "Introduce el lado: ";
            cin >> a;
            cout << "El area del cuadrado es de " << Cuadrado(a).calcularArea() << endl;
            break;
        case 5:
            cout << "Introduce el lado: ";
            cin >> a;
            cout << "El area del pentagono es de " << Pentagono(a).calcularArea() << endl;
            break;

        default:
            cout << "Opcion invalida.\n";
            return 1;
    }

    return 0;
}